### DISCUSSION TITLE
Understanding nested tag structure with multiple parent tags

### DISCUSSION THREAD

**Discussion by RagingRoosevelt** [TagStudioDev/TagStudio#485] [Q&A]
If I have a tag structure such as

```mermaid
classDiagram
  Actor <|-- TV
  Actor <|-- Movie
  TV <|-- Nathan Fillion
  Movie <|-- Nathan Fillion
```

How can I tag a file with only `Actor`←`TV`←`Nathan Fillion` without implying the `Actor`←`Movie`←`Nathan Fillion` tag?  Is the only way to have two `Nathan Fillion` tags: `Nathan Fillion (TV)` and `Nathan Fillion (Movie)`?

Having to make multiple child tags seems like it goes against some of the design philosophy of TagStudio.  It feels like it should be possible to chose which path on the tag tree you use to tag with.  Especially since the way that would display would be `Nathan Fillion (TV) (TV)`

**Comment by CyanVoxel** [comment]
I feel attaching `Nathan Fillion` to broad concepts such as `TV` or `Movie` in-between the concept of `Actor` is a misstep here. `Nathan Fillion` isn't directly a `TV` or a `Movie`, but he is an `Actor`. Here's how I might structure these tags to achieve the same effect of searching for `Nathan Fillion` in `TV` or `Movies` separately:

`Actor`←`Nathan Fillion`
`TV`←`Firefly`
`Movie`←`Guardians of the Galaxy Vol. 3`
*On an image of Malcom Reynolds aboard the Serenity:* `Nathan Fillion (Actor)`, `Firefly (TV)`

Searching for `Nathan Fillion` will return any images/videos/files with him as usual, but now searching for both `Nathan Fillion` and `TV` should only return files of him staring in TV shows. Same goes for any `Movies`, `Video Games`, etc. tagged in the same manner.

**Reply by RagingRoosevelt** [reply]
hmmmm. yeah, that's fair for my example.  Perhaps another example might be a better illustration of how it could be useful:

```mermaid
classDiagram
  Piercing <|-- Nose
  Piercing <|-- Ear
  Ear <|-- Barbell
  Nose <|-- Barbell
  Ear <|-- Captive Bead Ring
  Nose <|-- Captive Bead Ring
```

Technically you could make the same organization argument, turning it into:

```mermaid
classDiagram
  Piercing Location <|-- Nose
  Piercing Location <|-- Ear
  Jewlrey Type <|-- Captive Bead Ring
  Jewlrey Type <|-- Barbell
```

And then you could tag an image with `Nose (Piercing Location)` and `Captive Bead Ring (Jewlrey Type)`.  The difficulty arises when I'm trying to tag an image where someone has a CBR in their ear and a Barbell in their nose.  In the first tag structure, I could tag with `Piercing`←`Nose`←`Barbell` and `Piercing`←`Ear`←`Captive Bead Ring` and this would be distinct from, for example `Piercing`←`Ear`←`Barbell`

The second tag structure would only let me tag with `Piercing Location`←`Nose`, `Jewlrey Type`←`Captive Bead Ring`, etc which doesn't capture the full information.


I guess another possible solution is to just have multiple tags with the same name.  I guess I should maybe be thinking of tags as a tree and not a graph, since if I want to tag something with a particular path on a graph, that's essentially just a branch of a tree and my tags should reflect that.
